tests/test_deployments/test_app.py

```diff
@@ -248,30 +249,38 @@ def get_logs() -> str:
                 lines = "".join(f.readlines())
             return lines
 
-        def get_missing(messages: List[str]) -> List[str]:
-            i, max_retries = 0, 5
-            while messages and i < max_retries:
-                i += 1
-                time.sleep(1)
-                messages = [line for line in messages if line not in get_logs()]
+        def get_missing(messages: Set[str]) -> Set[str]:
+            while messages:
+                logs = get_logs()
+                messages -= {line for line in messages if line in logs}
```

Comment:
change introduced because of this error observed on Windows 3.8 CI run: https://github.com/valory-xyz/open-autonomy/actions/runs/3306777482/jobs/5457971216

```
================================== FAILURES ===================================
_______________ TestTendermintLogMessages.test_tendermint_logs ________________

self = <tests.test_deployments.test_app.TestTendermintLogMessages object at 0x0000018CC66360A0>

    @wait_for_node_to_run
    def test_tendermint_logs(self) -> None:
        """Test Tendermint logs"""
    
        def get_logs() -> str:
            with open(os.environ["LOG_FILE"], "r") as f:
                lines = "".join(f.readlines())
            return lines
    
        def get_missing(messages: List[str]) -> List[str]:
            i, max_retries = 0, 5
            while messages and i < max_retries:
                i += 1
                time.sleep(1)
                messages = [line for line in messages if line not in get_logs()]
            return messages
    
        before_stopping = [
            "Tendermint process started",
            "Monitoring thread started",
            "Starting multiAppConn service",
            "Starting localClient service",
            "This node is a validator",
        ]
    
        after_stopping = [
            "Monitoring thread terminated",
            "Tendermint process stopped",
        ]
    
>       assert not get_missing(before_stopping), get_logs()
E       AssertionError: Tendermint process started
E         Monitoring thread started
E         I[2022-10-23|11:06:50.159] Starting multiAppConn service                module=proxy impl=multiAppConn
E         I[2022-10-23|11:06:50.159] Starting localClient service                 module=abci-client connection=query impl=localClient
E         I[2022-10-23|11:06:50.159] Starting localClient service                 module=abci-client connection=snapshot impl=localClient
E         I[2022-10-23|11:06:50.159] Starting localClient service                 module=abci-client connection=mempool impl=localClient
E         I[2022-10-23|11:06:50.159] Starting localClient service                 module=abci-client connection=consensus impl=localClient
E         I[2022-10-23|11:06:50.159] Starting EventBus service                    module=events impl=EventBus
E         I[2022-10-23|11:06:50.159] Starting PubSub service                      module=pubsub impl=PubSub
E         
E       assert not ['This node is a validator']
E        +  where ['This node is a validator'] = <function TestTendermintLogMessages.test_tendermint_logs.<locals>.get_missing at 0x0000018CC33C8AF0>(['Tendermint process started', 'Monitoring thread started', 'Starting multiAppConn service', 'Starting localClient service', 'This node is a validator'])

tests\test_deployments\test_app.py:272: AssertionError
```